Update create_graph.py

seigodev opened the pull request:
Implementation of the hierarchical

seigodev committed:
Update create_graph.py

seigodev committed:
add files

demo_sample.json is create_graph.py results.
example.html is each node links.

seigodev commented:
add result json_file, and add html_file and js_file to view the graph

seigodev committed:
Update create_graph.py

Add Node class.
Edit all function in this file.  Rename variables , function name and so on.

seigodev committed:
Update create_graph.py

Rewrite create_node_list().

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+        href: ノードのリンク。str()。
+        is_dummy: ノードがダミーか否か。bool()。
+    """
+    def __init__(self, name, target_nodes, source_nodes, x, y, href, is_dummy):
```
name以外はデフォルト値をセットできたほうが良いでしょう

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+        return f"name: {name}, target_nodes: {target_nodes}, source_nodes: {source_nodes}, (x, y)= ({x}, {y})"
+
+
+def create_node_list():
```
引数が抜けています

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+    node_dict = {}
+    # node_dict, node_listの作成
+    for k, v in input_node_dict.items():
+        n = Node(k, set(), set(), -1, -1, v[1], False)
```
Nodeのコンストラクトのデフォルト引数

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+        インスタンス化されたノードのリスト。
+    """
+    node_list = []
+    node_dict = {}
```
keyとvalueの説明を書いてください．
名前はname2nodeのほうが良かったかもしれません．

aabaa replied:
関数名ではなく，変数 node_dict を変更してください．
node_dict -> name2node
です

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+            dの順番をランダムにしたもの
+        """
+        keys = list(d.keys())
+        random.shuffle(keys)
```
シャッフルは一回で十分

aabaa commented on demo/create_graph.py:
```diff
@@ -8,33 +8,135 @@
+        random.shuffle(keys)
+        [(key, d[key]) for key in keys]
+        random.shuffle(keys)
+        keys = [(key, d[key]) for key in keys]
```
型が変わるので，keysではなく違う名前の変数を用意するかダイレクトに
return dict([(key, d[key]) for key in keys])
と書く

aabaa commented on demo/draw_graph.js:
```diff
@@ -0,0 +1,73 @@
+    $.getJSON("./demo_sample.json", function(graph_data) {
+        //描画(graph_draw()をここに書き写す)
+        // cytoscapeグラフの作成(初期化)
+        var cy = cytoscape({
```
基本的にvarは使わずletを使ってください

seigodev committed:
Update create_graph.py

Fix requested changes .
・Nodeクラスのデフォルト引数の追加 [24行]
・create_node_list() → name2node(input_node_dict) [42行]
・コンストラクタのデフォルト引数による変更 [67行]
・key, valueの説明の追加 [64-81行]
・shuffle_dict(d)のシャッフル回数、returnの値の変更 [93-103行]

seigodev committed:
Update draw_graph.js

Convert 'var' to 'let'.

seigodev commented:
Fix these change requests.

aabaa reviewed (changes requested):
関数名ではなく，変数 node_dict を変更してください．
node_dict -> name2node
です

seigodev committed:
Fix create_graph.py

node_dict -> name2node
